EM.Sparql

```diff
@@ -0,0 +1,313 @@
+
+Title: Identifying Classes with Multiple Sub-properties in CCO
+Query Description: This query retrieves classes in the Common Core Ontology (CCO) that have multiple sub-properties. 
+It ensures that these properties are distinct and belong to a specific domain, highlighting potential complexities in class definitions.
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+
+SELECT ?class ?property1 ?property2
+WHERE {
+  # Selecting all classes in CCO
+  ?class a owl:Class .
+
+  # Retrieve the first sub-property of the class
+  ?property1 rdfs:subPropertyOf ?class .
+  
+  # Retrieve a second distinct sub-property of the same class
+  ?property2 rdfs:subPropertyOf ?class .
+
+  # Ensure that the two properties are distinct and belong to a specific domain
+  FILTER(?property1 != ?property2)
+
+  # Optional: Filtering based on specific characteristics of properties (e.g., a given domain)
+  # Here we can filter for properties that are defined within a certain namespace, which could indicate a type or category
+  FILTER(CONTAINS(STR(?property1), "http://example.org/specific")) 
+  FILTER(CONTAINS(STR(?property2), "http://example.org/specific")) 
+}
+
+11. Identifying Classes Without Labels (Level 4 - 10 points)
+
+Title: Missing Class Label Detection
+Constraint Description: Identifies classes that lack human-readable labels.
+Severity: Warning
+
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?class
+WHERE {
+  ?class a rdfs:Class .
+  FILTER NOT EXISTS { ?class rdfs:label ?label }
+}
+
+This query seeks to surface classes that lack human-readable labels. 
+Labels are vital for enabling users to understand the ontology’s structure, and their absence may hamper usability.
+
+12. Detecting Missing Annotations (Labels or Comments) (Level 4 - 10 points)
+
+Title: Missing Annotation Detection
+Constraint Description: Identifies ontology elements missing both labels and comments.
+Severity: Warning
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?element
+WHERE {
+  ?element a ?type .
+  FILTER(?type IN (rdf:Property, rdfs:Class, owl:ObjectProperty, owl:DatatypeProperty))
+  FILTER NOT EXISTS { ?element rdfs:label ?label }
+  FILTER NOT EXISTS { ?element rdfs:comment ?comment }
+}
+
+This query identifies ontology elements missing critical annotations, such as labels or comments. 
+These annotations provide context and clarify the purpose of each element, which is essential for effective ontology utilization.
+
+13. Detecting Properties Shared Across Multiple Classes (Level 3 - 20 points)
+
+Title: Shared Property Analysis
+Constraint Description: Identifies properties used by instances of multiple classes.
+Severity: Warning
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
```

Comment:
Compare this against the following and report back the results:

PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?property ?class1 ?class2
WHERE {
  ?property a rdf:Property .
  
  ?instance1 a ?class1 .
  ?instance1 ?property ?value .
  
  ?instance2 a ?class2 .
  ?instance2 ?property ?value .
  
  FILTER(?class1 != ?class2)
}